Sources/App/Controllers/API/TodoController.swift

```diff
@@ -0,0 +1,33 @@
+import Vapor
+import FluentSQLite
+
+/// Controls basic CRUD operations on `Todo`s.
+final class TodoController {
+    /// Returns a list of all `Todo`s.
+    func index(_ req: Request) throws -> Future<[Todo]> {
+        let repository = try req.make(TodoRepository.self)
+
+        var criteria: [FilterOperator<Todo.Database, Todo>] = []
+
+        if let title = req.query[String.self, at: "title"] {
+            criteria.append(.make(\Todo.title, .equal, [title]))
+        }
+
+        return repository.findBy(criteria: criteria, orderBy: nil, on: req)
+    }
+
+    /// Saves a decoded `Todo` to the database.
+    func create(_ req: Request) throws -> Future<Todo> {
+        return try req.content.decode(Todo.self).flatMap { todo in
+            let repository = try req.make(TodoRepository.self)
+            return repository.save(model: todo, on: req)
+        }
+    }
+
+    /// Deletes a parameterized `Todo`.
+    func delete(_ req: Request) throws -> Future<HTTPStatus> {
+        return try req.parameters.next(Todo.self).flatMap { todo in
+            return todo.delete(on: req)
+        }.transform(to: .ok)
```

Comment:
I think this should be `noContent` (204): https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html